packages/bson-transpilers/symbols/php/templates.yaml

```diff
@@ -0,0 +1,1237 @@
+            return literal.toLowerCase();
+        }
+    IntegerTypeTemplate: &IntegerTypeTemplate null # args: literal, argType (i.e. '_string', '_decimal' etc)
+    DecimalTypeTemplate: &DecimalTypeTemplate null # args: literal, argType
+    LongBasicTypeTemplate: &LongBasicTypeTemplate null # args: literal, argType
+    HexTypeTemplate: &HexTypeTemplate null # args: literal, argType
+    OctalTypeTemplate: &OctalTypeTemplate !!js/function >
+        (literal) => {
+            let offset = 0;
+
+            if (
+                literal.charAt(0) === '0' &&
+                (literal.charAt(1) === '0' || literal.charAt(1) === 'o' || literal.charAt(1) === 'O')
+            ) {
+                offset = 2;
+            } else if (literal.charAt(0) === '0') {
+                offset = 1;
+            }
+
+            literal = `0${literal.substr(offset, literal.length - 1)}`;
+
+            return literal;
+        }
+    NumericTypeTemplate: &NumericTypeTemplate null # args: literal, argType
+    ArrayTypeTemplate: &ArrayTypeTemplate !!js/function >
+        (literal, depth) => {
+            if (literal === '') {
+                return '[]'
+            }
+            return `[${literal}]`;
+        }
+    ArrayTypeArgsTemplate: &ArrayTypeArgsTemplate null # Args: single array element, nestedness, lastElement? (note: not being used atm)
+    NullTypeTemplate: &NullTypeTemplate !!js/function >
+        () => {
+            return 'null';
+        }
+    UndefinedTypeTemplate: &UndefinedTypeTemplate !!js/function >
+        () => {
+            return 'null';
+        }
+    ObjectTypeTemplate: &ObjectTypeTemplate !!js/function >
+        (literal) => {
+            return `${literal}`;
+        }
+    ObjectTypeArgsTemplate: &ObjectTypeArgsTemplate !!js/function >
+        (args, depth) => {
+            let isObjectCastRequired = true;
+        
+            if (args.length === 0) {
+                return `(object) []`;
+            }
+
+            const isExpectedIndex = (actualIndex, expectedIndex) => {
+                let parsedActualIndex = parseInt(actualIndex);
+                return !(isNaN(parsedActualIndex) || ('' + parsedActualIndex !== actualIndex) || parsedActualIndex != expectedIndex);
+            }
+        
+            let indexTest = 0;
+            let pairs = args.map((arg) => {
+                if (isObjectCastRequired && !isExpectedIndex(arg[0], indexTest)) {
+                    isObjectCastRequired = false;
+                }
+                indexTest++;
+                return `${this.utils.stringifyWithSingleQuotes(arg[0])} => ${arg[1]}`;
+            }).join(', ');
+
+            // Rebuilding pairs for numeric sequential indexes without quotes
+            if (isObjectCastRequired) {
+                pairs = args.map((arg) => {
+                    return `${arg[0]} => ${arg[1]}`;
+                }).join(', ');
+            }
+
+            return `${isObjectCastRequired ? '(object) ' : ''}[${pairs}]`
```

Comment:
I'm not sure why GitHub is rendering the syntax oddly here, but perhaps it has to do with failure to recognize the backtick as the start of a string template (singe `return` is also green).

In any event, you might want to add a semicolon after this line, as you have elsewhere. OK to resolve after changing.

Reply:
Semicolon was added.